apps/www/app/api-v2/ticket-og/route.tsx

```diff
@@ -0,0 +1,555 @@
+      await supabaseAdminClient
+        .from(LW_TABLE)
+        .update({ shared_on_twitter: 'now' })
+        .eq('launch_week', 'lw12')
+        .eq('username', username)
+        .is('shared_on_twitter', null)
+    } else if (userAgent?.toLocaleLowerCase().includes('linkedin')) {
+      await supabaseAdminClient
+        .from(LW_TABLE)
+        .update({ shared_on_linkedin: 'now' })
+        .eq('launch_week', 'lw12')
+        .eq('username', username)
+        .is('shared_on_linkedin', null)
+    }
+
+    // Get ticket data
+    const { data: user, error } = await supabaseAdminClient
+      .from(LW_MATERIALIZED_VIEW)
+      .select(
+        'id, name, ticket_number, shared_on_twitter, shared_on_linkedin, platinum, secret, role, company, location'
+      )
+      .eq('launch_week', 'lw12')
+      .eq('username', username)
+      .maybeSingle()
+
+    if (error) console.log('fetch error', error.message)
+    if (!user) throw new Error(error?.message ?? 'user not found')
+
+    const {
+      name,
+      ticket_number: ticketNumber,
+      secret,
+      platinum: isPlatinum,
+      shared_on_twitter: sharedOnTwitter,
+      shared_on_linkedin: sharedOnLinkedIn,
+    } = user
+
+    console.log(user)
+
+    const platinum = isPlatinum ?? (!!sharedOnTwitter && !!sharedOnLinkedIn) ?? false
+    if (assumePlatinum && !platinum)
+      return await fetch(`${STORAGE_URL}/assets/platinum_no_meme.jpg`)
+
+    // Generate image and upload to storage.
+    const ticketType = secret ? 'secret' : platinum ? 'platinum' : 'regular'
+
+    const STYLING_CONFIG = {
+      BACKGROUND: themes[ticketType].OG_BACKGROUND,
+      FOREGROUND: themes[ticketType].TICKET_FOREGROUND,
+      FOREGROUND_LIGHT: themes[ticketType].TICKET_FOREGROUND_LIGHT,
+      TICKET_BORDER: themes[ticketType].TICKET_BORDER,
+      TICKET_FOREGROUND: themes[ticketType].TICKET_FOREGROUND,
+      TICKET_BACKGROUND: themes[ticketType].TICKET_BACKGROUND,
+      TICKET_BACKGROUND_CODE: themes[ticketType].TICKET_BACKGROUND_CODE,
+      TICKET_FOREGROUND_LIGHT: themes[ticketType].TICKET_FOREGROUND_LIGHT,
+      BORDER: themes[ticketType].TICKET_BORDER,
+      CODE_LINE_NUMBER: themes[ticketType].CODE_LINE_NUMBER,
+      CODE_BASE: themes[ticketType].CODE_THEME['hljs'].color,
+      CODE_HIGHLIGHT: themes[ticketType].CODE_HIGHLIGHT_BACKGROUND,
+      CODE_FUNCTION: themes[ticketType].CODE_THEME['hljs'].color,
+      CODE_VARIABLE: themes[ticketType].CODE_THEME['hljs'].color,
+      CODE_METHOD: themes[ticketType].CODE_THEME['hljs'].color,
+      CODE_EXPRESSION: themes[ticketType].CODE_THEME['hljs-keyword'].color,
+      CODE_STRING: themes[ticketType].CODE_THEME['hljs-string'].color,
+      CODE_NUMBER: themes[ticketType].CODE_THEME['hljs'].color,
+      CODE_NULL: themes[ticketType].CODE_THEME['hljs'].color,
+      JSON_KEY: themes[ticketType].CODE_THEME['hljs-attr'].color,
+    }
+
+    const fontData = await font
+    const monoFontData = await mono_font
```

Comment:
```suggestion
    const [fontData, monoFontData] = await Promise.all([font, mono_font])
```

don't know that it'll make a difference in practice, but maybe?